### PR TITLE
[java] jaxrs (all) add usetags option

### LINKED ISSUE #5937: [BUG] [jaxrs-spec] Cannot change names of API classes (no useTags option and apiNameSufix/apiNamePrefix don't work)
openapi-generator-maven-plugin 4.3.0
Generator jaxrs-spec

I haven't found any way to change class name for APIs generated by openapi-generator-maven-plugin with jaxrs-spec generator.

For example, if I have endpoint /my-api/v1/foo, the generator is going to always create MyApiApi class.

The generator does not have useTags option, as for example Spring generator.
The generator also does not take into account additional properties apiNameSuffix and apiNamePrefix, when they are defined as: 
```
<additionalProperties>
     <additionalProperty>apiNameSuffix=RestApi</additionalProperty>
      <additionalProperty>apiNamePrefix=Prefix</additionalProperty>
</additionalProperties>
```

When I have multiple resources, for example /my-api/v1/foo and /my-api/v1/bar, there is no option to put them into separate API classes (FooV1Api and BarV1Api).

##### Suggest a fix
I suggest to:
1. Add useTags option to this generator
2. Add apiNameSuffix and apiNamePrefix support for it

### PR DESCRIPTION
**Pull request by PhilippParis**
### Changes
- fixes #5937
- Adds "useTags" option to all jaxrs code generators (AbstractJAXRSServerCodegen):
  - jaxrs-jersey
  - jaxrs-spec
  - jaxrs-cxf-cdi
  - jaxrs-resteasy
  - jaxrs-cxf
  - jaxrs-cxf-extended
  - java-msf4j
  - jaxrs-resteasy-eap
- jaxrs-spec
  - changed handling of root paths: e.g "/:", "/{id}:" to simplify code and create a more consistent behaviour
    - old: use tag for classname
    - new: use tag only if useTags is enabled, use "DefaultApi" if not
- @Path class level annotation
  - for all generators above and the microprofile generator (uses same jaxrs postprocessing)
  - extended the "commonPath" to contain more than only the root path if possible
  - e.g. "/group1/subgroup1/op1" -> "/group1/subgroup1" is moved to class level annotation

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

### CODE REVIEW

**Comment by tatarco** [comment]
Its' Amazing and super needed, Thank you!

**Comment by ruedigerk** [comment]
I would really love to see this getting merged.

**Comment by tatarco** [comment]
> I would really love to see this getting merged.

so am i, it's really missing from jaxrs-spec

**Comment by famod** [comment]
@PhilippParis Thanks, my team also needs this. Any chance to resolve the conflicts?

@wing328 & @jimschubert I guess this comes too late for 5.0.0?

**Comment by PhilippParis** [comment]
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

any chance this is getting reviewed & merged?

**Comment by kdebski85** [comment]
@wing328 Could you please review?

**Review by jimschubert — APPROVED** [review]
Awesome contribution. Sorry for such a delay in the review with the backlog of PRs and the preparations for 5.0.0 release.

I think to match how other generators work, we should default to `useTags=true`… but I'd like to get this in to avoid any additional merge conflicts. I'll open another PR for changing the default value so we can discuss more and have fewer potential merge conflicts.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaJAXRSServerCodegen.java [hunk 1]
```diff
@@ -281,5 +313,8 @@ public void setUseBeanValidation(boolean useBeanValidation) {
         this.useBeanValidation = useBeanValidation;
     }
 
-
+    @VisibleForTesting
```
Comment (jimschubert) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaJAXRSServerCodegen.java:316]: `VisibleForTesting` is only for testing purposes and should probably be removed in favor of a package-private visibility.

**Comment by jimschubert** [comment]
I just checked the `useTags` again, and I think the defaults provided here make sense. We can get some feedback from the community on the initial 5.0.0 release.

**Pull request merged**